design/open/pull-request.md

```diff
@@ -6,23 +6,30 @@ The pull-request captures the merge operation we request to apply.
 ### Goals
 
 - Capture request to merge a branch or specific reference
-- Mechanism to review, discuss and approve changes before applying to a branch
+- Mechanism to review, discuss and merge changes before applying to a branch
 - Automation using actions can verify / provide feedback on pull-request
 
 ### How it will work
 
 Introduce a new resource called pull-request at the repository level.
-The pull-request will hold owner (the user who created the PR), assignee, reviewers the source branch/reference and the target branch.
-Each reviewer can set approved/reject (or clear his response) on the PR.
+The pull-request will hold owner (the user who created the PR), description, assignee, reviewers the source branch/reference and the target branch.
+Each reviewer can set approved/reject (or clear his response) on the PR and add comments to the PR discussion. Comments associated to a PR will a flat ordered list of posted comments on the PR level.
+New comments are appended to the end of the discussion and will include a time stamp, commenter and the text with the comment.
 PR can be merge operation is enabled when at least one reviewer approves or when no reviewers are assigned.
 PR can be closed at any time.
-PR will not be updated when new commits into the source branch.
+PR will not be updated with changes to the source or destination branches, like updates to objects, commits, merges, etc. 
 Merge PR will perform a merge operation and merge the PR as merged. If a conflict or an error occurs while merging, it will be returned as a response to the request.
 When a repository is deleted, all pull-requests will be deleted with it.
 
+Actions - no change in current support with pre/post merge. These actions will be triggered when the PR is merged.
+A new event called 'pull-request' will be triggered when is created (or by request). These action will be triggered if no conflict found on an internal commit that will be visible to the action in order to verify the PR data.
+
+
 ### Other features to consider
 
 - PR auto-update with latest branch commits. PR will include the base commit and all the changes until the latest commit on branch.
 - Multiple reviews with text from each reviewer. Can be even per level of commit the current PR addresses.
 - New hook can be trigger when PR is updated. Action triggered by PR will be considered as reviewers and the status is approved/rejects based on a successful run.
+- Do we need to support merge dry run to enable UI indication for conflicts?
```

Comment:
This can be done automatically, simply diff the 2 and notify on conflict. btw I think that the UI will present the diff and will show conflicts anyhow

Reply:
Make sense I'll remove this item, thanks.